src/main/java/com/indeed/jiraactions/ActionFactory.java

```diff
@@ -129,15 +167,63 @@ private long timeInState(final Action prevAction, final History history) {
     }
 
     private long timeInState(final Action prevAction, final DateTime changeTimestamp) {
-        if(!Objects.equals(prevAction.getPrevstatus(), prevAction.getStatus())) {
+        if (!Objects.equals(prevAction.getPrevstatus(), prevAction.getStatus())) {
             return getTimeDiff(prevAction.getTimestamp(), changeTimestamp);
         }
 
         return getTimeDiff(prevAction.getTimestamp(), changeTimestamp) + prevAction.getTimeinstate();
     }
 
+    private long getDateResolved(final String resolutionDate) {
+        if (resolutionDate == null || resolutionDate.equals("")) {
+            return 0;
+        } else {
+            return Long.parseLong(resolutionDate.split("T")[0].replaceAll("[^\\d]", ""));
+        }
+    }
+
+    private long getDateClosed(final Action prevAction, final History history) {
+        final String status = history.itemExist("status") ? history.getItemLastValue("status") : prevAction.getStatus();
+        if (status.equals("Closed")) {
```

Comment:
Not sure it matters here, but consider inverting this:
`if ("Closed".equals(status)) {`

I know it reads weird, but it's null-safe. `null.equals("Closed")` will throw a NullPointerException, but `"Closed".equals(null)` will just evaluate to false.